update-website.py

```
@@ -0,0 +1,814 @@
+                        # Extract the redirect URL
+                        match = re.search(r'url=([^"\'>\s]+)', content)
+                        if match:
+                            redirect = match.group(1)
+                            if redirect.startswith('/#'):
+                                # It's a section in the index page
+                                return f"{base_url}{redirect[1:]}"  # Remove the leading /
+                            elif redirect.startswith('/'):
+                                return f"{base_url}{redirect}"
+                            else:
+                                return f"{base_url}/{redirect}"
+                except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)